invenio_records_resources/services/records/results.py

```diff
@@ -279,6 +292,10 @@ def add_service_value(self, service, value):
 
     def add_dereferenced_record(self, service, value, resolved_rec):
         """Save the dereferenced record."""
+        if resolved_rec is None:
+            resolved_rec = self.defaults({"id": value})
+            # mark the record as a "ghost" record
+            resolved_rec["is_ghost"] = True
```

Comment:
I left it here intentionally, so it's kind of forced instead of relying on each concrete implementation to add it. That way, we can eventually rely on this key to be there in all different views...